### DISCUSSION TITLE
pyvista widgets can only work in Jupyter "toggle rendering mode (remote)" not "toggle rendering mode (local)"

### DISCUSSION THREAD

**Discussion by YongcaiHuang** [pyvista/pyvista#6028] [Q&A] [answered]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
import pyvista as pv
from pyvista import examples

mesh = examples.download_nefertiti()
p = pv.Plotter()
p.add_mesh_clip_box(mesh, color='white')
p.show(cpos=[-1, -1, 0.2], jupyter_backend="trame")
```

### Description

I am new using this wonderful project. below is my question.

Why do pyvista widgets only work in Jupyter "toggle rendering mode (remote)" not "toggle rendering mode (local)". The reason why is that I found "toggle rendering mode (remote)" is less resolution (kind of blurred), but I can drag widgets to interact. when I change to "toggle rendering mode (local)", the resolution gets higher, but widgets cannot be dragged (what I can do is rotation). Is there any way to interact with widgets in "local" mode?

https://github.com/pyvista/pyvista/assets/97007177/8445d430-566f-4574-8b32-54639660028f

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Sat May 04 13:43:54 2024 CST

                  OS : Darwin
              CPU(s) : 10
             Machine : arm64
        Architecture : 64bit
                 RAM : 16.0 GiB
         Environment : Jupyter
         File system : apfs
          GPU Vendor : Apple
        GPU Renderer : Apple M2 Pro
         GPU Version : 4.1 Metal - 88
    MathText Support : True

  Python 3.10.0 | packaged by conda-forge | (default, Nov 20 2021, 02:27:15)
  [Clang 11.1.0 ]

             pyvista : 0.43.6
                 vtk : 9.3.0
               numpy : 1.26.4
          matplotlib : 3.8.4
              scooby : 0.9.2
               pooch : 1.8.1
              pillow : 10.3.0
             imageio : 2.34.1
             IPython : 8.24.0
          ipywidgets : 8.1.2
               scipy : 1.13.0
          jupyterlab : 4.1.8
               trame : 3.6.0
        trame_client : 3.0.2
        trame_server : 3.0.0
           trame_vtk : 2.8.7
       trame_vuetify : 2.4.3
jupyter_server_proxy : 4.1.2
        nest_asyncio : 1.6.0
--------------------------------------------------------------------------------
```

**Comment by jourdain** [comment, marked as answer]
Widgets are not supported in local mode. Also the resolution can be increased when using remote rendering. I just don't know how to set it within PyVista.

**Reply by YongcaiHuang** [reply]
I see. thanks for the comments! It would be great to see widgets interaction support in local mode in the future if possible.

**Reply by jourdain** [reply]
That is the plan with WASM support but it will take some time as additional communication will be required to keep the server in sync with local interaction.
